StRoot/StFcsWaveformFitMaker/StFcsWaveformFitMaker.cxx

```diff
@@ -620,6 +615,45 @@ TGraphAsymmErrors* StFcsWaveformFitMaker::makeTGraphAsymmErrors(StFcsHit* hit){
     return gae;
 }
 
+
+float StFcsWaveformFitMaker::AnaPed( TGraphAsymmErrors* g, float& ped, float& pedstd )
+{
+  int n = g->GetN();
+  double *t = g->GetX();
+  double *a = g->GetY();    
+  int p=0;
+  double sumsq = 0;//For variance
+  for(int i=0; i<n; i++){
+    int tb1=t[i];
+    if(mPedMin<=tb1 && tb1<=mPedMax){    
+      p+=a[i];
+      sumsq += a[i]*a[i];
+    }
+  }
+  ped = float(p)/(mPedMax-mPedMin+1.0);
+  pedstd = sqrt( ( sumsq-((double(p)*double(p))/(mPedMax-mPedMin+1.0)) )/(mPedMax-mPedMin) );//Variance/StdDev using naive algorithm from https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
+  return ped;
+}
+
+float StFcsWaveformFitMaker::AnaPed( StFcsHit* hit, float& ped, float& pedstd )
+{
+  int p=0;
+  int n = hit->nTimeBin();
+  double sumsq = 0;//For variance
+  for(int i=0; i<n; i++){
+    int tb=hit->timebin(i);
+    if(mPedMin<=tb && tb<=mPedMax ){
+      int adc = hit->adc(i);
+      p+=adc;
+      sumsq += adc*adc;
+    }
+    if(tb>=mPedMax) break;
+  }
+  ped = float(p)/(mPedMax-mPedMin+1.0);
+  pedstd = sqrt( ( sumsq-((double(p)*double(p))/(mPedMax-mPedMin+1.0)) )/(mPedMax-mPedMin) );//Variance/StdDev using naive algorithm from https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance
```

Comment:
You don't need to cast `p` to double. It will be done automatically using the C++ rules for implicit type casting.

Reply:
done. Is is this because of the 1.0 in the denominator?

Reply:
yes, all values are converted to double in arithmetic expressions